farey/algo.py

```diff
@@ -19,15 +19,35 @@ def find_rational_approximation(x, places=7):
     if x == 0:
         return Rational(0, 1)
     elif x < 0:
-        return -find_rational_approximation(-x, places)
+        return -find_rational_approximation(-x, method, places, max_denominator)
     elif x >= 1:
-        return int(x // 1) + find_rational_approximation(x % 1, places)
+        return int(x // 1) + find_rational_approximation(
+            x % 1, method, places, max_denominator
+        )
     elif 0.5 < x < 1:
-        return 1 - find_rational_approximation(1 - x, places)
+        return 1 - find_rational_approximation(1 - x, method, places, max_denominator)
 
-    # we only need to look in the range between (0, 1)
-    # and (1, n) for n = math.floor(1 / x), since we will always reach
-    # this eventually by the Farey addition
+    if method == "farey":
+        if places is not None and max_denominator is None:
+            return _farey_algorithm_accuracy(x, places)
+        if places is None and max_denominator is not None:
+            return _farey_algorithm_denominator(x, max_denominator)
+        else:
+            print("Must specify one of decimal places or maximum denominator")
+            return None
+
+
+def _farey_algorithm_accuracy(x, places=7):
```

Comment:
We should have `0 < x <= 0.5` here, but since it's a private method and we only pass `x` in this range, no need to include an additional constraint here.